.github/e2e-tests.yml

```diff
@@ -0,0 +1,128 @@
+# This file specifies the GitHub runner for each E2E test and is utilized by all E2E CI workflows.
+#
+# Each entry in this file includes the following:
+# - The GitHub runner (runs-on field) that will execute tests.
+# - The tests that will be run by the runner.
+# - The workflows (e.g., Run PR E2E Tests, Run Nightly E2E Tests) that should trigger these tests.
+#
+runner-test-matrix:
+
+  # Example of 1 runner for all tests in integration-tests/smoke/ocr_test.go
+  - id: smoke/ocr_test.go:* 
+    path: integration-tests/smoke/ocr_test.go
+    test-type: docker
+    runs-on: ubuntu-latest
+    workflows:
+      - Run PR E2E Tests
+      - Run Nightly E2E Tests
+    test-cmd: cd integration-tests/ && go test smoke/ocr_test.go -timeout 30m -count=1 -json
+    pyroscope-env: ci-smoke-ocr-evm-simulated-nightly
+
+  # Example of 2 separate runners for the same test file but different tests. Can be used if tests if are too heavy to run on the same runner
+  - id: smoke/ocr2_test.go:^TestOCRv2Request$
+    path: integration-tests/smoke/ocr2_test.go
+    test-type: docker
+    runs-on: ubuntu-latest
+    workflows:
+      - Run PR E2E Tests
+      - Run Nightly E2E Tests
+    test-cmd: cd integration-tests/ && go test smoke/ocr2_test.go -test.run ^TestOCRv2Request$ -test.parallel=1 -timeout 30m -count=1 -json
```

Comment:
wonder if that's explicitly stated somewhere (where?) that it's up to the person defining a workflow to make sure that `id` matches `test-cmd`? Do we have any check that will fail if it doesn't? What will happen if there's a mismatch? Say that someone id'ed this one as `smoke/ocr2_test.go:*`, would that make our filtering tool falsely assume that all tests in that file are running somewhere? Would there be any other side-effect?